pyk/src/pyk/kast/outer_parser.py

```diff
@@ -62,20 +63,28 @@ class OuterParser:
     _lexer: Iterator[Token]
     _la: Token
     _la2: Token
+    _source: Path | None
 
-    def __init__(self, it: Iterable[str]):
+    def __init__(self, it: Iterable[str], source: Path | None = None):
         self._lexer = outer_lexer(it)
         self._la = next(self._lexer)
         self._la2 = next(self._lexer, _EOF_TOKEN)
+        self._source = source
 
     def _consume(self) -> str:
         res = self._la.text
         self._la, self._la2 = self._la2, next(self._lexer, _EOF_TOKEN)
         return res
 
+    def _error_location_string(self, t: Token) -> str:
+        if not self._source:
+            return ''
+        return f'{self._source}:{t.loc.line}:{t.loc.col}: '
+
     def _match(self, token_type: TokenType) -> str:
         if self._la.type != token_type:
-            raise ValueError(f'Expected {token_type.name}, got: {self._la.type.name}')
+            location_string = self._error_location_string(self._la)
+            raise ValueError(location_string + f'Expected {token_type.name}, got: {self._la.type.name}')
```

Comment:
Consider extracting a function:

```python3
def _unexpected_token(token: Token, expected_types: Iterable[TokenType] = ()) -> ValueError:
    ...
```

This I think covers all three modified cases.